tests/AgentTests.hs

```diff
@@ -302,16 +302,36 @@ testMsgDeliveryServerRestart t alice bob = do
   where
     withServer test' = withSmpServerStoreLogOn (ATransport t) testPort2 (const test') `shouldReturn` ()
 
--- testServerConnectionAfterError :: Transport c => TProxy c -> c -> IO ()
--- testServerConnectionAfterError t bob = do
---   withAgent $ \alice -> do
---     withServer $ do
---       connect (bob, "bob") (alice, "alice")
+testServerConnectionAfterError :: forall c. Transport c => TProxy c -> [c] -> IO ()
+testServerConnectionAfterError t _ = do
+  withAgent1 $ \bob -> do
+    withAgent2 $ \alice -> do
+      withServer $ do
+        connect (bob, "bob") (alice, "alice")
 
---     bob <# ("", "alice", DOWN)
---     alice #: ("2", "bob", "SEND 11\nhello again") #> ("2", "bob", MID 5)
---     alice #:# "nothing else delivered before the server is restarted"
---     bob #:# "nothing else delivered before the server is restarted"
+      bob <# ("", "alice", DOWN)
+      alice <# ("", "bob", DOWN)
+      alice #: ("1", "bob", "SEND 5\nhello") #> ("1", "bob", MID 4)
+      alice #:# "nothing else delivered before the server is restarted"
+      bob #:# "nothing else delivered before the server is restarted"
+
+  withAgent1 $ \bob -> do
+    withAgent2 $ \alice -> do
+      bob #: ("1", "alice", "SUB") #> ("1", "alice", ERR (BROKER NETWORK))
+      alice #: ("1", "bob", "SUB") #> ("1", "bob", ERR (BROKER NETWORK))
+      withServer $ do
+        bob <# ("", "alice", UP)
+        alice <# ("", "bob", SENT 4)
+        alice <# ("", "bob", UP)
+        bob <#= \case ("", "alice", Msg "hello") -> True; _ -> False
+        alice #: ("1", "bob", "SEND 11\nhello again") #> ("1", "bob", MID 5)
+        alice <# ("", "bob", SENT 5)
```

Comment:
test bob receives this message as well?
bob <#= \case ("", "alice", Msg "hello again") -> True; _ -> False